### ISSUE TITLE
Icon texture does not update on talent or gear swap

### ISSUE THREAD

**Issue by Hemario** [Sidoine/Ovale#749]
**Describe the bug**
Icon texture does not update for spells when they get replaced by another spell due to talent swaps. 
Icon texture of trinket does not change when you swap trinket.

**To Reproduce**
Custom script to test (protection paladin)
```
Define(blessed_hammer_talent 23469) 
Define(blessed_hammer 204019)
    SpellInfo(blessed_hammer cd=6)
Define(hammer_of_the_righteous 53595)
    SpellInfo(hammer_of_the_righteous cd=6)
    SpellRequire(hammer_of_the_righteous replaced_by set=blessed_hammer enabled=(hastalent(blessed_hammer_talent)))

AddIcon {
    spell(hammer_of_the_righteous)
}
AddIcon {
    Item(Trinket0Slot)
}
AddIcon {
    Item(Trinket1Slot)
}
```
1. Change talents from Blessed Hammer to Holy Shield
2. Change your trinkets

**Expected behavior**
The icon textures should update according to the newly selected talent or trinket.
After a /reload the item texture is correct.

**Comment by ultijlam** [comment]
I've seen this as well when I wear two on-use trinkets and I swap the locations of the two trinkets.

**Comment by Hemario** [comment]
Even worse for Havoc Demonhunters (after merging https://github.com/Hemario/Ovale/commit/b2fd91db8c7657f1f1657c25dae5ae9b4359a65c)


```
Include(ovale_common)
Include(ovale_demonhunter_spells)

AddIcon {
spell(chaos_strike_havoc)
}

AddIcon {
Spell(blade_dance)
}

```

Casting meta will not change the texture of the icons

**Comment by ultijlam** [comment]
Here is an even simpler example:
```
AddCheckBox(test_checkbox "test")

AddIcon {
    if CheckBoxOn(test_checkbox) 1
    0
}
```
Using this script, Ovale will show "0.0" in the icon frame, but if you toggle the checkbox, it doesn't change to "1.0" unless you reload the UI.  I've turned on debugging for OvaleCompile and it does say:
```
OvaleCompile: Ovale_CheckBoxValueChanged: advance age to 4.
OvaleCompile: Script has changed. Evaluating...
```

**Comment by ultijlam** [comment]
I've figured out why this is happening, and I will have a pullup for this issue shortly.

**Comment by ultijlam** [comment]
@Hemario I think this is fixed in Ovale 9.0.34. Can you verify?
